app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -286,22 +286,33 @@ class NavigationOverlayFragment : Fragment() {
         ).map { channel -> channel.forwardRemoveEventsToRepo() }
     }
 
-    private fun observeShouldDisplayPinnedTiles(): Disposable {
-        // We considered putting all common channel behavior into one reusable function (and may
-        // still in the future), but were concerned about a potential problem.
+    private fun observeChannelVisibility(): List<Disposable> {
+        // NOTE: for unknown reasons, removing the last tile in a channel crashes if it is visible.
```

Comment:
Ideally, we'd cover this with a test too but I know we're short on time and this test would be hard to write so I'm okay without it.

Reply:
Yeah, and without knowing the root cause this would have to be a UI test, too.